spec/unit/plugins/linux/network_spec.rb

```diff
@@ -1335,6 +1335,59 @@
         end
       end
 
+      describe "when there's a source field in a local route entry but no dev" do
+        let(:linux_ip_route) do
+          <<~EOM
+            default proto bird src 172.16.19.39 metric 10 \\      nexthop via 10.1.81.1 dev eth0 weight 20
+            10.116.201.0/24 dev eth0  proto kernel  src 10.116.201.76
+            192.168.0.0/24 dev eth0  proto kernel  src 192.168.0.2
+            default via 10.116.201.254 dev eth0 metric 20 src 10.116.201.74
+          EOM
+        end
+        let(:linux_ip_route_inet6) { "" }
+
+        it "maps the no-dev route and sets ipaddress" do
```

Comment:
what do you mean by no-dev? That route has a dev... every route here has a dev.

Reply:
@jaymzh if you look at line 1341 above, there is no "dev" shown which is what we're attempting to assert.

Reply:
what ... are you talking about. It's a multipath route that explicit;y has "dev eth0" in the nexthop.

Reply:
devices associated with multi-hop routes aren't considered 'the device associated with the route' ; that logic exists as is today - unrelated to this PR